#166265858 user can request a book

PelumiAlesh opened the pull request:
### What does this PR do?
Create endpoint for requesting a new book
#### Description of Task to be completed?
Create a new table for Requested books
Create an endpoint for requesting new books
Add ISBN column to Book's table
#### How should this be manually tested?
Clone the repo
Run `npm i`
Run `npm run start`
Make a `POST api/v1/Books/request`
#### Any background context you want to provide?
I refactored some codes
#### What are the relevant pivotal tracker stories?
[#166265858](https://www.pivotaltracker.com/story/show/166265858 "Users should be able to request for a book that's not in the library")
#### Screenshots (if appropriate)
![Screenshot 2019-06-10 at 5 55 58 PM](https://user-images.githubusercontent.com/29557902/59211943-0ddf9080-8ba9-11e9-81bb-94c3d3ce4ac8.png)
#### Questions:

PelumiAlesh committed:
feature(books): user can request book
>
> refactor test code
> write test for requesting books
> create request book endpoint
> update docs

[Finishes #166265858]

emp-daisy commented on server/controllers/bookController.js:
```diff
@@ -7,18 +7,18 @@ const log = debug('dev');
         authorID,
-        status,
-        year
+        active,
```
What's the reason for the switch in name?

PelumiAlesh replied:
The status was representing whether the book is **active** or not, and the value was a _boolean_. So I decided to switch the name to **active** so _boolean_ would make sense( **active**: _true_ ) better than ( **status**: _true_ )

emp-daisy replied:
What do you mean active or not? Same as borrowed or not?

I feel it would have to be improved so we are able to store the status from an ENUM option.
The status could be either available, borrowed, under review etc.
Changing the active or inactive is limiting the options

emp-daisy commented on server/routes/bookRoute.js:
```diff
@@ -8,5 +8,6 @@ bookRoute.post('/', Validate.addBook, BookController.addBook);
 bookRoute.get('/', BookController.getAllBooks);
 bookRoute.get('/:id', Validate.id, BookController.getSpecificBook);
 bookRoute.delete('/:id', Validate.id, BookController.deleteBook);
+bookRoute.post('/request',  Validate.requestBook, BookController.requestBook)
```
Is this route open to the public or only authenticated users?

PelumiAlesh replied:
It has no authentication yet.

encodedBicoding replied:
I think anyone should be able to request a book. Because it help the company know what books people are looking for.

emp-daisy replied:
Read the story description again

OvieMudi replied:
The story specified "authenticated user"

emp-daisy commented:
Why was `server/db/seeders/20190606210612-demo-books.js` deleted?

PelumiAlesh commented:
> Why was `server/db/seeders/20190606210612-demo-books.js` deleted?

Kafilat wrote another demo-books file that uses faker js. So my `server/db/seeders/20190606210612-demo-books.js` was no longer needed

emp-daisy commented:
@PelumiAlesh Any update on this?

PelumiAlesh committed:
bg(auth): add authentication to book request

add authentication to book request route
edit docs to reflect new updates on books
